dspace-api/src/main/java/org/dspace/content/dao/impl/MetadataFieldDAOImpl.java

```diff
@@ -79,6 +94,30 @@ public MetadataField findByElement(Context context, MetadataSchema metadataSchem
     @Override
     public MetadataField findByElement(Context context, String metadataSchema, String element, String qualifier)
         throws SQLException {
+        String key = metadataSchema + "." + element + "." + qualifier;
+        if (cachedFields.containsKey(key)) {
```

Comment:
It might be good to add an inline comment here to describe the logic you've added, so that we don't forget why this was done. Something like:

```
// Check for this field in cached list of IDs
// If found in cache, then this field has been previously accessed
// we should be able to load it from Hibernate's session cache (instead of DB)
// This provides a large speed increase as MetadataField is accessed frequently
```